packages/knip/src/plugins/aws-cdk/index.ts

```diff
@@ -0,0 +1,38 @@
+import { toDependency } from 'src/util/input.js';
+import type { IsPluginEnabled, Plugin, ResolveConfig } from '../../types/config.js';
+import { compact } from '../../util/array.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { AwsCdkConfig } from './types.ts';
+
+// https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib-readme.html
+
+const title = 'aws-cdk';
+
+const enablers = ['aws-cdk'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ["cdk.json"];
+
+const resolveConfig: ResolveConfig<AwsCdkConfig> = async config => {
+  if (!config) return [];
+
+  const app = config.app.split(" ")[0];
+  const watch = config.watch?.include ?? [];
```

Comment:
My guess would be that the `watch` values aren't necessarily dependencies, nor entry files. Maybe we should ignore those. Maybe there are other options that represent entry points?

Reply:
To the best of my knowledge, the `watch.include` entries are file that should form the codebase for the IaC when using AWS CDK. From the name, it would indicate those are source files being **watched** for changes when running `cdk watch` which would allow an immediate changes of the generated CloudFormation template.

Reply:
If we would add the `watch.include` patterns, then they should be returned as `project` files (not `entry` files), i.e. `toInput('bin/**/*.ts')`. Adding "all" files as entry files would have two disadvantages: by default, unused exports of entry files are not reported, and Knip wouldn't be able to find unused files (as entry files are inherently used files).

However, the default `project` pattern is `**/*.{js,ts}`, so adding more specific patterns from a plugin wouldn't be effective.

My suggestion would be to ignore the `watch.include` patterns and see whether the defaults Knip finds are sufficient, then we could add default entry patterns in the plugin if there are any, and perhaps there are specific entry patterns to be found in aws-cdk configuration files.

Reply:
Ah ok that makes sense. I'll remove that 'watch' from the returned list.

Reply:
Addressed in 7efb03dbe2d9175efed5647b28610bbd0f2be98b